### ISSUE TITLE
New Setting : enable/disable graph tree history 

### ISSUE THREAD

**Issue by sysres-dev** [Cacti/cacti#2540]
**Is your feature request related to a problem? Please describe.**
Our cacti users complained about some behaviour changes in the interface, we used cacti 0.8.8 for a long time, and they would like some behaviors closer to it :
- on the graph tree, when you close the page and come back again, it opens back the branches where it were previously opened, since we have 27k devices splitted in hundreds of tree and subtrees, the user would prefer to start again on a clean tree, without history when the page open again.

**Describe the solution you'd like**
It would be very nice to have new settings in global setting panel to choose cacti's behavior regarding this.

**Describe alternatives you've considered**
For now we solved thoses problems by hardcoding values in right places in the code : 
```

tree history clean :
+++ cacti/lib/html_tree.php
150,153
>         window.onunload = function() {
>                 localStorage.removeItem('graph_tree_history');
>         }


```
**Additional context**
Here are some stats about my Cacti installation:

Cacti Version | 1.1.38
Cacti OS | unix
RSA Fingerprint | 22:9a:ef:ad:d8:de:1d:e7:95:7d:89:98:bd:4f:b9:df
NET-SNMP Version | NET-SNMP version: 5.7.2
RRDtool Version | RRDtool 1.4.x
Devices | 26897
Graphs | 458451
Data Sources:
Script/Command: 16597
SNMP Get: 76962
SNMP Query: 291955
Script Query: 62267
Script Server: 26697
Script Query - Script Server: 31729
Total: 506207

**Comment by netniV** [comment]
Deep linking of the graph tree was added in 1.2, so this may have been resolved at that point.

**Comment by netniV** [comment]
Sorry, I mis-read what you were after.  Yes, we did correct it to maintain the history properly.  If you would prefer not to have that functionality, you are correct it should become both a global setting and a user setting.

Can you create a pull request with the appropriate changes?  You would need to add the user setting to `$settings_user` array in `includes/global_settings.php`.  Once you have that, you can then read that setting using `read_user_setting('setting name')` to determine if you wipe the history.  Default should be off (eg, keep the history)

**Comment by sysres-dev** [comment]
Implemented and merged in PR #2552, i'm closing the issue.
